lib/beaker/command_factory.rb

```diff
@@ -16,7 +16,11 @@ def assertions
     end
 
     def execute(command, options={}, &block)
-      result = self.exec(Command.new(command), options)
+      cmd_create_options = {}
+      exec_opts = options.dup
+      cmd_create_options[:prepend_cmds] = exec_opts.delete(:prepend_cmds) || nil
```

Comment:
I added this comment earlier today but not under this pull request.  Adding it again here.



@anodelman While running my tests I stumbled on a problem. This line is actually modifies the the variable that stores my options outside of this method. The quick irb session below displays the problem.

1) First define a variable 'options' that contains my options including :prepend_cmds

irb(main):001:0> options = {:prepend_cmds => "foo bar baz", :option2 => "option"}
=> {:prepend_cmds=>"foo bar baz", :option2=>"option"}
irb(main):002:0> puts(options)
{:prepend_cmds=>"foo bar baz", :option2=>"option"}
=> nil

2) Now define a quick method to process the options and remove :prepend_cmds if it exists and store in a new var.

irb(main):003:0> def process(options={})
irb(main):004:1> new_opts = {}
irb(main):005:1> new_opts[:prepend_cmds] = options.delete(:prepend_cmds) || nil
irb(main):006:1> end
=> nil

3) Quick test to make sure my options var outside the process method still contains :prepend_cmds

irb(main):007:0> puts(options)
{:prepend_cmds=>"foo bar baz", :option2=>"option"}
=> nil

4) Now call the process method on my options

irb(main):008:0> process(options)
=> "foo bar baz"

5) Unexpectedly (IMO) my variable 'options' outside the method is modified.
irb(main):009:0> puts(options)
{:option2=>"option"}
=> nil

This problem is causing my prepended commands to be stripped in the following case:

opts = {:pty => true, :prepend_cmds => pp(agent)}
agent.install_package(package, '', nil, opts)
agent.check_for_package("puppet", opts)

When I call agent.install_package my opts are intact, but when I then call agent.check_for_package the :prepend_cmds have been removed causing the command to fail.